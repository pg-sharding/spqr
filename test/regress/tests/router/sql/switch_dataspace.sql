\c spqr-console

CREATE DISTRIBUTION ds1;
CREATE DISTRIBUTION ds2;

CREATE SHARDING RULE r1 COLUMNS w_id FOR DISTRIBUTION ds1;
CREATE SHARDING RULE r2 COLUMNS w_id FOR DISTRIBUTION ds2;

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh2 FOR DISTRIBUTION ds1;

CREATE KEY RANGE krid3 FROM 11 ROUTE TO sh2 FOR DISTRIBUTION ds2;

\c regress

SET __spqr__distribution = ds1;

CREATE TABLE xx (w_id int);

INSERT INTO xx(w_id) VALUES(5);

INSERT INTO xx(w_id) VALUES(20);

SELECT * FROM xx WHERE w_id=5;

SET __spqr__distribution = ds2;

SELECT * FROM xx WHERE w_id=5;

SET __spqr__distribution = "default";

DROP TABLE xx;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP SHARDING RULE ALL;
DROP KEY RANGE ALL;