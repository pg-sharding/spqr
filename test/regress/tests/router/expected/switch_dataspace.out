\c spqr-console

		SQPR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DATASPACE ALL CASCADE;
 drop dataspace 
----------------
(0 rows)

CREATE DATASPACE ds1;
         add dataspace         
-------------------------------
 created dataspace with id ds1
(1 row)

CREATE DATASPACE ds2;
         add dataspace         
-------------------------------
 created dataspace with id ds2
(1 row)

ADD SHARDING RULE r1 COLUMNS w_id FOR DATASPACE ds1;
                                    add sharding rule                                    
-----------------------------------------------------------------------------------------
 created sharding rule r1 for table (*) with columns [w_id, hash: x->x] in ds1 dataspace
(1 row)

ADD SHARDING RULE r2 COLUMNS w_id FOR DATASPACE ds2;
                                    add sharding rule                                    
-----------------------------------------------------------------------------------------
 created sharding rule r2 for table (*) with columns [w_id, hash: x->x] in ds2 dataspace
(1 row)

ADD KEY RANGE krid1 FROM 1 TO 11 ROUTE TO sh1 FOR DATASPACE ds1;
         add key range          
--------------------------------
 created key range from 1 to 11
(1 row)

ADD KEY RANGE krid2 FROM 11 TO 31 ROUTE TO sh2 FOR DATASPACE ds1;
          add key range          
---------------------------------
 created key range from 11 to 31
(1 row)

ADD KEY RANGE krid3 FROM 11 TO 31 ROUTE TO sh2 FOR DATASPACE ds2;
          add key range          
---------------------------------
 created key range from 11 to 31
(1 row)

\c regress
SET __spqr__DATASPACE = ds1;
CREATE TABLE xx (w_id int);
NOTICE: send query to shard(s) : sh1,sh2
INSERT INTO xx(w_id) VALUES(5);
NOTICE: send query to shard(s) : sh1
INSERT INTO xx(w_id) VALUES(20);
NOTICE: send query to shard(s) : sh2
SELECT * FROM xx WHERE w_id=5;
NOTICE: send query to shard(s) : sh1
 w_id 
------
    5
(1 row)

SET __spqr__DATASPACE = ds2;
SELECT * FROM xx WHERE w_id=5;
ERROR:  client proccessing error: error processing query 'SELECT * FROM xx WHERE w_id=5;': too complex query to parse, tx status IDLE
